torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1320,10 +1320,13 @@ Tensor binary_cross_entropy_target_backward(
   const Tensor& target,
   const c10::optional<Tensor>& weight,
   int64_t reduction) {
-  auto grad_target = (1. - self).log_().sub_(self.log()).mul_(grad);
+  // test_fn_grad_nn_functional_binary_cross_entropy_cpu_float64 fails if
+  // aten::mul_ is used here.
+  auto grad_target = (1. - self).log_().sub_(self.log());
+  grad_target = grad_target * grad;
```

Comment:
Can you give a bit more info on why this fails when `mul_` is used?

I could imagine this changing the perf of the function (although probably trivially).

Reply:
@Lezcano suggested to replace the problematic inline operation when i asked about it. ofc it's possible that i misunderstood since i'm new to the codebase. so all errors are mine :)

test output without the fix:

```
torch.autograd.gradcheck.GradcheckError: While computing batched gradients, got: vmap: aten::mul_(self, *extra_args) is not possible because there exists a Tensor `other` in extra_args that has more elements than `self`. This happened due to `other` being vmapped over but `self` not being vmapped over at level 1. Please try to use out-of-place operators instead of aten::mul_. If said operator is being called inside the PyTorch framework, please file a bug report instead.

gradcheck or gradgradcheck failed while testing batched gradient computation.
This could have been invoked in a number of ways (via a test that calls
gradcheck/gradgradcheck directly or via an autogenerated test).

If you are adding a new operator, please file an issue and then use one of the
workarounds. The workaround depends on how your test invokes gradcheck/gradgradcheck.
If the test
- manually invokes gradcheck/gradgradcheck, then call gradcheck/gradgradcheck
  with `check_batched_grad=False` as a keyword argument.
- is OpInfo-based (e.g., in test_ops.py), then modify the OpInfo for the test
  to have `check_batched_grad=False` and/or `check_batched_gradgrad=False`.

If you're modifying an existing operator that supports batched grad computation,
or wish to make a new operator work with batched grad computation, please read
the following.

To compute batched grads (e.g., jacobians, hessians), we vmap over the backward
computation. The most common failure case is if there is a 'vmap-incompatible
operation' in the backward pass. Please see
NOTE: [How to write vmap-compatible backward formulas]
in the codebase for an explanation of how to fix this.
```

Reply:
I think that the comment in the code is not even necessary.
The tl;dr is that our derivatives should be differentiable and in-place functions are not differentiable. In-place functions do not play well with functorch either.
In this case, the perf hit is minimal really. We use this sort of pattern in most of the codebase. It could be even less if one moves out of the tensors once they're never used again to deallocate them, but we barely ever implement this sort of pattern really.